ceph_deploy/util/pkg_managers.py

```diff
@@ -319,6 +330,16 @@ class DNF(RPMManagerBase):
     executable = 'dnf'
     name = 'dnf'
 
+    def install(self, packages, **kw):
+        extra_install_flags = kw.pop('extra_install_flags', [])
+        if '--best' not in extra_install_flags:
```

Comment:
So yes, Yum automatically updates packages, and DNF does not.

Doing the following on CentOS 7:

```
ceph-deploy install --dev next cent701
ceph-deploy install --dev master cent701
```

does an upgrade, even though the underlying commands are just `yum -y install ceph ceph-radosgw` in each case.

Adding `--best` seems to recreate this behavior.  Of course DNF is new and there may be consequences to `--best` we do not understand yet.  The docs mention that you are more likely to get into dependency resolution problems because you are telling it to always use the newer version of the package, even if it can't resolve the deps.

If that happens, that probably means there's a problem with the packages, IMO.  We'll see!